wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -116,94 +116,96 @@ public static function assignChapterGroupToIndividual(string $op, string $object
-    $collectionCampEntries = $decodedData['Eck_Collection_Camp1'] ?? [];
-    if (empty($collectionCampEntries)) {
+    if (!$newStatus) {
       return;
     }
 
-    foreach ($collectionCampEntries as $entry) {
-      $collectionCampData = $entry['fields'] ?? NULL;
-
-      // Access the subtype.
-      $subtypeId = $collectionCampData['subtype'] ?? NULL;
-      if ($subtypeId === NULL) {
-        continue;
-      }
-
-      // Access the id within the decoded data.
-      $campId = $collectionCampData['id'] ?? NULL;
-      if ($campId === NULL) {
-        continue;
-      }
-
-      // Fetch the collection camp details.
-      $collectionCamps = EckEntity::get('Collection_Camp', FALSE)
-        ->addWhere('id', '=', $campId)
-        ->setLimit(1)
-        ->execute();
-
-      if (empty($collectionCamps)) {
-        continue;
-      }
-
-      $collectionCampsCreatedDate = $collectionCamps->first()['created_date'] ?? NULL;
-
-      // Get the year.
-      $year = date('Y', strtotime($collectionCampsCreatedDate));
-
-      // Fetch the state ID.
-      $stateId = self::getStateIdForSubtype($collectionCampData, $subtypeId);
-
-      if (!$stateId) {
-        continue;
-      }
-
-      // Fetch the state abbreviation.
-      $stateProvinces = StateProvince::get(FALSE)
-        ->addWhere('id', '=', $stateId)
-        ->setLimit(1)
-        ->execute();
-
-      if (empty($stateProvinces)) {
-        continue;
-      }
+    $collectionCamp = EckEntity::get('Collection_Camp', FALSE)
+      ->addSelect('Collection_Camp_Core_Details.Status', 'Collection_Camp_Core_Details.Contact_Id', 'title')
+      ->addWhere('id', '=', $objectId)
+      ->execute()->single();
 
-      $stateAbbreviation = $stateProvinces->first()['abbreviation'] ?? NULL;
+    $currentStatus = $collectionCamp['Collection_Camp_Core_Details.Status'];
 
-      if (!$stateAbbreviation) {
-        continue;
+    // Check for status change.
+    if ($currentStatus !== $newStatus) {
+      if ($newStatus === 'authorized') {
+        // Access the subtype.
+        $subtypeId = $objectRef['subtype'] ?? NULL;
+        if ($subtypeId === NULL) {
+          return;
+        }
+
+        // Access the id within the decoded data.
+        $campId = $objectRef['id'] ?? NULL;
+        if ($campId === NULL) {
+          return;
+        }
+
+        // Fetch the collection camp details.
+        $collectionCamps = EckEntity::get('Collection_Camp', FALSE)
+          ->addWhere('id', '=', $campId)
+          ->setLimit(1)
+          ->execute()->single();
+
+        if (empty($collectionCamps)) {
+          return;
+        }
+
+        $collectionCampsCreatedDate = $collectionCamps['created_date'] ?? NULL;
+
+        // Get the year.
+        $year = date('Y', strtotime($collectionCampsCreatedDate));
+
+        // Fetch the state ID.
+        $stateId = self::getStateIdForSubtype($objectRef, $subtypeId);
+
+        if (!$stateId) {
+          return;
```

Comment:
Can we add a log here so that if the code is not generated for any camp due to this? If Goonj teams comes to us and says that a collection camp does not have code then we can check the logs